src/services/rpc-provider/rpc-provider.service.ts

```diff
@@ -40,6 +44,34 @@ export default class RpcProviderService {
     });
     return new JsonRpcBatchProvider(rpcUrl);
   }
+
+  public async getTimeTravelBlock(period: TimeTravelPeriod): Promise<number> {
+    const currentBlock = await rpcProviderService.getBlockNumber();
+    const blocksInDay = Math.round(twentyFourHoursInSecs / this.blockTime);
+
+    switch (period) {
+      case '24h':
+        return currentBlock - blocksInDay;
+      default:
+        return currentBlock - blocksInDay;
+    }
+  }
+
+  public get blockTime(): number {
+    switch (networkId.value) {
+      case Network.MAINNET:
+        return 13;
+      case Network.POLYGON:
+        return 2;
+      case Network.ARBITRUM:
+        return 3;
+      case Network.KOVAN:
+        // Should be ~4s but this causes subgraph to return with unindexed block error.
+        return 1;
+      default:
+        return 13;
+    }
+  }
```

Comment:
I get why you've put this in here, but the rpc service should be exclusively about dealing with rpcs. This logic should either go in the web3 service, a new block service or maybe just a composable.